chat/QuickJump.vue

```diff
@@ -0,0 +1,459 @@
+<template>
+  <div class="quick-jump-overlay" v-show="visible" @click="hide">
+    <div class="quick-jump-container" @click.stop>
+      <div class="quick-jump-header">
+        <span class="fas fa-search"></span>
+        <span>{{ l('quickJump.title') }}</span>
+      </div>
+      <input
+        ref="searchInput"
+        v-model="searchQuery"
+        :placeholder="l('quickJump.placeholder')"
+        class="quick-jump-input"
+        @keydown="onKeyDown"
+        @input="onInput"
+      />
+      <div class="quick-jump-results" v-show="filteredResults.length > 0">
+        <div
+          v-for="(result, index) in filteredResults"
+          :key="result.key"
+          :class="['quick-jump-result', { selected: index === selectedIndex }]"
+          @click="selectResult(result)"
+          @mouseenter="selectedIndex = index"
+        >
+          <span class="result-icon">
+            <span
+              v-if="result.type === 'channel'"
+              class="fas fa-hashtag"
+            ></span>
+            <span
+              v-else-if="result.type === 'private'"
+              class="far fa-user-circle"
+            ></span>
+            <span v-else class="fas fa-home"></span>
+          </span>
+          <span class="result-name">{{ result.name }}</span>
+          <span class="result-description" v-if="result.description">
+            {{ result.description }}
+          </span>
+        </div>
+      </div>
+      <div
+        class="quick-jump-footer"
+        v-show="filteredResults.length === 0 && searchQuery.length > 0"
+      >
+        <span>{{ l('quickJump.noResults') }}</span>
+        <div
+          class="quick-jump-new-conversation"
+          :class="{ selected: selectedIndex === -1 }"
+          @click="openNewConversation"
+          @mouseenter="selectedIndex = -1"
+        >
+          <span class="result-icon">
+            <span class="fas fa-plus"></span>
+          </span>
+          <span class="result-name">{{
+            l('quickJump.openNewConversation', searchQuery)
```

Comment:
Probably a good idea to make this look similar to the selected items, that way people will know that pressing Enter works here too.